core/src/main/scala/kafka/coordinator/transaction/ProducerIdManager.scala

```diff
@@ -70,7 +70,7 @@ case class ProducerIdBlock(brokerId: Int, blockStartId: Long, blockEndId: Long)
   }
 }
 
-class ProducerIdManager(val brokerId: Int, val zkClient: KafkaZkClient) extends Logging {
+class ProducerIdManager(val brokerId: Int, val zkClient: KafkaZkClient) extends ProducerIdMgr with Logging {
```

Comment:
I realize this is just a first cut, but can we have the trait be named something like `ProducerIdGenerator`?  It's confusing to have both Manager and Mgr.
Also, it would be nice if the `ProducerIdGenerator` were defined in this file rather than elsewhere